# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsIGlobalObject.h
# Commit: 78181c32bdc6
# Full Hash: 78181c32bdc6b1d66eac04f3ce11a2902f477823
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-02-27 21:52:52
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Add the setup for WebTaskSchedulingState r=jjaschke,dom-core,arai
#   
#   Spec: https://wicg.github.io/scheduling-apis/#scheduling-state
#   
#   This patch add the setup for WebTaskSchedulingState so that it
# ==============================================================================

diff -r 912cf1adac80 -r 78181c32bdc6 dom/base/nsIGlobalObject.h
--- a/dom/base/nsIGlobalObject.h	Thu Feb 27 16:17:34 2025 +0000
+++ b/dom/base/nsIGlobalObject.h	Thu Feb 27 16:17:35 2025 +0000
@@ -53,6 +53,7 @@
 class ServiceWorkerRegistration;
 class ServiceWorkerRegistrationDescriptor;
 class StorageManager;
+class WebTaskSchedulingState;
 enum class CallerType : uint32_t;
 }  // namespace dom
 namespace ipc {
@@ -194,6 +195,13 @@
     return nullptr;
   }
 
+  virtual void SetWebTaskSchedulingState(
+      mozilla::dom::WebTaskSchedulingState* aState) {}
+  virtual mozilla::dom::WebTaskSchedulingState* GetWebTaskSchedulingState()
+      const {
+    return nullptr;
+  }
+
   // For globals with a concept of a Base URI (windows, workers), the base URI,
   // nullptr otherwise.
   virtual nsIURI* GetBaseURI() const;